sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessStreamConsumer.java

```diff
@@ -21,47 +21,65 @@
 
 import java.io.BufferedReader;
 import java.io.IOException;
+import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.nio.charset.StandardCharsets;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
-import java.util.concurrent.atomic.AtomicBoolean;
+import org.sonar.api.Startable;
 import org.sonar.api.batch.BatchSide;
 import org.sonar.api.batch.ScannerSide;
 import org.sonar.api.utils.log.Logger;
 import org.sonar.api.utils.log.Loggers;
-import org.sonarsource.api.sonarlint.SonarLintSide;
 
 @BatchSide
 @ScannerSide
-@SonarLintSide
-public class ExternalProcessErrorConsumer {
+public class ExternalProcessStreamConsumer implements Startable {
 
-  private static final Logger LOG = Loggers.get(ExternalProcessErrorConsumer.class);
+  private static final Logger LOG = Loggers.get(ExternalProcessStreamConsumer.class);
+  private ExecutorService executorService;
 
-  public final void consumeStdError(Process process) {
-    final ExecutorService errorConsumer = Executors.newSingleThreadExecutor();
-    errorConsumer.submit(() -> {
-      try (BufferedReader errorReader = new BufferedReader(new InputStreamReader(process.getErrorStream(), StandardCharsets.UTF_8))) {
-        readErrors(errorReader);
+  public final void consumeStream(InputStream inputStream, StreamConsumer streamConsumer) {
+    executorService.submit(() -> {
+      try (BufferedReader errorReader = new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8))) {
+        readErrors(errorReader, streamConsumer);
       } catch (IOException e) {
-        LOG.error("Error while reading error stream", e);
+        LOG.error("Error while reading stream", e);
       }
     });
   }
 
-  protected void readErrors(BufferedReader errorReader) {
-    AtomicBoolean tsNotFound = new AtomicBoolean(false);
-    errorReader.lines().forEach(line -> {
-      if (line.contains("Error: Cannot find module 'typescript'")) {
-        tsNotFound.set(true);
-      }
+  protected void readErrors(BufferedReader errorReader, StreamConsumer streamConsumer) {
+    errorReader.lines().forEach(streamConsumer::consumeLine);
+    streamConsumer.finished();
+  }
 
-      LOG.error(line);
-    });
+  @Override
+  public void start() {
+    if (executorService == null) {
+      executorService = Executors.newCachedThreadPool(r -> {
+        Thread thread = new Thread(r);
+        thread.setName("sonarts-stream-consumer");
+        thread.setDaemon(true);
+        return thread;
+      });
+    }
+  }
+
+  @Override
+  public void stop() {
+    if (executorService != null && !executorService.isShutdown()) {
+      executorService.shutdown();
```

Comment:
After shutting down, it's probably a good idea to wait for termination (with a timeout). Otherwise, we might have logs out of sync:  they will arrive after the analysis has terminated. This should be done after closing the stream on the server side, otherwise it will wait for ever (shutdown doesn't do anything to the running thread).

Reply:
Thanks @dbmeneses . I added the call to `awaitTermination` with callback, but I am not sure what you mean about closing the streams. Normally they should be closed when process finishes, so I expect all the submitted tasks to terminate.

Reply:
Alright, all good then!